### ISSUE TITLE
passing Configure -Dprefix a literal backslash, space sequence (\ ) confuses Configure

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14660]
Migrated from [rt.perl.org#124345](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=124345) (status was 'open')

Searchable as RT124345$

**Comment by p5pRT** [comment]
### From @kentfredric

The attached build script is expected to work\, but does not\. It instead gets hung up in Configure&#8203;:



---
System manual is in /usr/share/man/man1\.

Perl5 has manual pages available in source form\.
If you don't want the manual sources installed\, answer 'none'\.
sed&#8203;: \-e expression \#1\, char 87&#8203;: unterminated \`s' command
 
Where do the main Perl5 manual pages \(source\) go? \(~name ok\)


---
Configure script in question is that shipped with perl\-5\.21\.10

**Comment by p5pRT** [comment]
### From @kentfredric

[bad.sh](https://rt-archive.perl.org/perl5/Ticket/Attachment/1340654/717016/bad.sh)

**Comment by p5pRT** [comment]
### From @tux

On Sat\, 18 Apr 2015 17&#8203;:37&#8203;:01 \-0700\, Kent Fredric \(via RT\)
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> The attached build script is expected to work\, but does not\. It
> instead gets hung up in Configure&#8203;:

Thanks for spotting\.

The next change is waiting to be merged after the release of 5\.22&#8203;:

commit a46a738f1910a2c28366530e4366aa38fe47fc96
Author&#8203;: H\.Merijn Brand \<h\.m\.brand@&#8203;xs4all\.nl>
Date&#8203;:   Sun Apr 19 11&#8203;:20&#8203;:06 2015 \+0200

&nbsp;    Forbid the use of a literal \\ in Configure \-D arguments


: The reasoning behind this change is that optdef\.sh gets sourced and
: the arguments\, like prefix\, are passed around in different quotation
: styles and subshells all around\, and the \(unescaped\) backslash will
: not be pertained in the final config\.sh

This change has already been adopted by the upstream meta/dist package

\-\- 
H\.Merijn Brand  http&#8203;://tux\.nl   Perl Monger  http&#8203;://amsterdam\.pm\.org/
using perl5\.00307 \.\. 5\.21   porting perl5 on HP\-UX\, AIX\, and openSUSE
http&#8203;://mirrors\.develooper\.com/hpux/        http&#8203;://www\.test\-smoke\.org/
http&#8203;://qa\.perl\.org   http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by jkeenan** [comment]
> ### From @Tux
> 
> On Sat, 18 Apr 2015 17​:37​:01 -0700, Kent Fredric (via RT) <perlbug-followup@​perl.org> wrote​:
> 
> > The attached build script is expected to work, but does not. It
> > instead gets hung up in Configure​:
> 
> Thanks for spotting.
> 
> The next change is waiting to be merged after the release of 5.22​:
> 
> commit [a46a738](https://github.com/Perl/perl5/commit/a46a738f1910a2c28366530e4366aa38fe47fc96) Author​: H.Merijn Brand <h.m.brand@​xs4all.nl> Date​: Sun Apr 19 11​:20​:06 2015 +0200
> 
>   Forbid the use of a literal \ in Configure -D arguments
> 
> : The reasoning behind this change is that optdef.sh gets sourced and : the arguments, like prefix, are passed around in different quotation : styles and subshells all around, and the (unescaped) backslash will : not be pertained in the final config.sh
> 
> This change has already been adopted by the upstream meta/dist package
> 
> -- H.Merijn Brand http​://tux.nl Perl Monger http​://amsterdam.pm.org/ using perl5.00307 .. 5.21 porting perl5 on HP-UX, AIX, and openSUSE http​://mirrors.develooper.com/hpux/ http​://www.test-smoke.org/ http​://qa.perl.org http​://www.goldmark.org/jeff/stupid-disclaimers/

@Tux, was the patch above ever brought into Perl 5 `Configure`?  If not, do you wish to pursue it further?

**Comment by Tux** [comment]
This has been merged as
```
commit f41cd3a9d49c13be551a0daeb6aad1c3b9786d4b
Author: H.Merijn Brand - Tux <h.m.brand@xs4all.nl>
Date:   Sun Apr 19 10:56:12 2015 +0200

    Inhibit the use of \ in -D arguments
```
And amended in
```
commit e9a6b2913204856e44ea6134cb2ea95566830a39
Author: H.Merijn Brand - Tux <h.m.brand@xs4all.nl>
Date:   Thu Apr 21 15:17:23 2016 +0200

    Stuff missed in previous commits
```
It ended up in the core with these commits:
```
commit e244340e4067bb332773529bbff797bd14f103de
Author: Jarkko Hietaniemi <jhi@iki.fi>
Date:   Wed Dec 16 20:19:03 2015 -0500

    Configure: grep -q is not portable

    It does not work in SysV (solaris) or old BSD greps.
```
```
commit e6704a40e007cf599527cad0675d63c2ca3e423d
Author: H.Merijn Brand <h.m.brand@xs4all.nl>
Date:   Mon Jul 6 16:02:15 2015 +0200

    Regenerate Configure after backports and small fixes
```
